xobjects/context_cupy.py

```diff
@@ -387,13 +387,14 @@ def nplike_array_type(self):
     def linked_array_type(self):
         return LinkedArrayCupy
 
-    def __init__(self, default_block_size=256, device=None):
+    def __init__(self, default_block_size=256, default_shared_mem_size_bytes=0, device=None):
```

Comment:
Need to add a test to the test suite for the usage of the shared memory

Reply:
added test